.github/workflows/docker.yml

```diff
@@ -20,67 +18,70 @@ jobs:
       - name: Log in to GHCR
         run: |
           echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
-      - name: Push latest image
-        run: |
-          docker push "$CORE_IMAGE:latest"
-          docker tag "$CORE_IMAGE:latest" "$CORE_IMAGE_MIRROR:latest"
-          docker push "$CORE_IMAGE_MIRROR:latest"
-      - name: Push tagged image
-        if: "contains(github.ref, 'refs/tags')"
-        run: |
-          VERSION="$(grep -Eo "[0-9]+\.[0-9]+\.[0-9]+" common/lib/dependabot/version.rb)"
-          docker tag "$CORE_IMAGE:latest" "$CORE_IMAGE:$VERSION"
-          docker push "$CORE_IMAGE:$VERSION"
-          docker tag "$CORE_IMAGE:latest" "$CORE_IMAGE_MIRROR:$VERSION"
-          docker push "$CORE_IMAGE_MIRROR:$VERSION"
+      - name: Docker tags
+        id: tags
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            dependabot/dependabot-core
+            ghcr.io/dependabot/dependabot-core
+          flavor: |
+            latest=false
+          tags: |
+            type=raw,value=latest
+            type=semver,pattern={{version}}
+      - name: Build and push dependabot-core image
+        uses: docker/build-push-action@v3.1.1
+        with:
+          push: true
+          platforms: linux/amd64,linux/arm64
+          cache-from: type=registry,ref=dependabot/dependabot-core:latest
+          cache-to: type=inline
+          tags: ${{ steps.tags.outputs.tags }}
   push-development-image:
     name: Push dependabot-core-development image to GHCR
     runs-on: ubuntu-latest
     permissions:
       contents: read
       packages: write
     needs: push-core-image
-    env:
-      DEV_IMAGE: ghcr.io/dependabot/dependabot-core-development
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
-      - name: Build dependabot-core image
-        env:
-          DOCKER_BUILDKIT: 1
-        run: |
-          docker build \
-            -t "$DEV_IMAGE:latest" \
-            --build-arg BUILDKIT_INLINE_CACHE=1 \
-            --cache-from "$BASE_IMAGE" \
-            --cache-from "$CORE_IMAGE:latest" \
-            --cache-from "$DEV_IMAGE:latest" \
-            -f Dockerfile.development .
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2.0.0
       - name: Log in to GHCR
         run: |
           echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
-      - name: Push latest image
-        run: |
-          docker push "$DEV_IMAGE:latest"
-      - name: Push tagged image
-        if: "contains(github.ref, 'refs/tags')"
-        run: |
-          VERSION="$(grep -Eo "[0-9]+\.[0-9]+\.[0-9]+" common/lib/dependabot/version.rb)"
-          docker tag "$DEV_IMAGE:latest" "$DEV_IMAGE:$VERSION"
-          docker push "$DEV_IMAGE:$VERSION"
+      - name: Docker tags
```

Comment:
Metadata action correctly creates tags for docker images. It will automatically assign just the `latest` tag when workflow is triggered by push to default branch and will add semver version for tag pushes.